pallets/merkle-airdrop/src/lib.rs

```diff
@@ -101,27 +139,19 @@ pub mod pallet {
 
         /// Weight information for the extrinsics in this pallet.
         type WeightInfo: WeightInfo;
-    }
 
-    /// Storage for Merkle roots of each airdrop
-    #[pallet::storage]
-    #[pallet::getter(fn airdrop_merkle_roots)]
-    pub type AirdropMerkleRoots<T> = StorageMap<_, Blake2_128Concat, AirdropId, MerkleRoot>;
-
-    /// Storage for airdrop creators
-    #[pallet::storage]
-    #[pallet::getter(fn airdrop_creators)]
-    pub type AirdropCreators<T: Config> = StorageMap<_, Blake2_128Concat, AirdropId, T::AccountId>;
+        /// Block number provider.
+        type BlockNumberProvider: BlockNumberProvider<BlockNumber = BlockNumberFor<Self>>;
+    }
 
-    /// Storage for airdrop balances
+    /// Stores general info about an airdrop
     #[pallet::storage]
-    #[pallet::getter(fn airdrop_balances)]
-    pub type AirdropBalances<T: Config> = StorageMap<
+    #[pallet::getter(fn airdrop_info)]
+    pub type AirdropInfo<T: Config> = StorageMap<
         _,
         Blake2_128Concat,
         AirdropId,
-        <<T as Config>::Currency as Inspect<T::AccountId>>::Balance,
-        ValueQuery,
+        AirdropMetadata<BlockNumberFor<T>, BalanceOf<T>, T::AccountId>,
```

Comment:
I see now that we have a single map. In the previous implementation, we had separate maps to separate access to the different parts in different functions. I wonder if this is more optimal. Or is it not a problem to work with bigger maps?